src/web/mod.rs

```diff
@@ -167,6 +167,18 @@ impl Handler for MainHandler {
             }
         }
 
+        fn pass_iron_errors_with_redirect(e: IronError) -> IronResult<Response> {
```

Comment:
Is there a reason you made this a separate function instead of an inline closure like the other handler?

Reply:
I think it somehow came from the `if_404` handling being in a method. Looking at it now (and after some time), it actually makes more sense as a closure. 

I'll move it.